### ISSUE TITLE
Overwritten Authorization roles displayed

### ISSUE THREAD

**Issue by govza** [mattfrear/Swashbuckle.AspNetCore.Filters#144]
I'm trying to show Authorized roles information in my swagger UI.

Using this kind of structure in my Controller
```
[Authorize(Roles = "Admin,Employee")] // admin or employee
public class XController : Controller 
{
    [Authorize(Roles = "Admin")] // only admin
    public ActionResult ActionX() { ... }

    public ActionResult ActionX() { ... } // admin, employee
}
```
results to  (Auth roles: Admin,Employee, Admin) for first method.
So if my method controller Authorize is overwritten it's not respected, is it possible to fix that without overwriting the logic in controllers. It's really easy to have default one for all the controller and precise if needed in methods.

**Comment by mattfrear** [comment]
I didn't realise that the Authorize attributes worked that way.
Are you sure that ActionX in your example above is only Admin, and not also Employee?

**Comment by govza** [comment]
Yes it looks like it uses strategy as less restrictive rights applied on class level, and more restrictive to the method.
Also found out that it's not possible to add method level Authorize attribute with Role which is not applied in class-level.

`[Authorize(Roles = "Admin,Employee")] // admin or employee
public class XController : Controller 
{
    [Authorize(Roles = "Admin")] // only admin
    public ActionResult ActionX() { ... }

    public ActionResult ActionY() { ... } // admin, employee
    
    [Authorize(Roles = "Admin,Patient")]
    public ActionResult ActionZ() { ... } // admin but not patient
}`

**Comment by mattfrear** [comment]
OK, cool. I haven't used this filter in years, so you're welcome to fix it and submit a PR with tests.
